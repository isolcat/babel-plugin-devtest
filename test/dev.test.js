const babel = require("@babel/core");
const plugin = require("../");
const inputCode = `
        const a = 10;
        const b = 20;
        console.log(a+b)
        if(TEST){
            console.log("Hello World")
        }
        `;

describe("babel-plugin-devTest", () => {
    it("dev", () => {
        // input
        process.env.NODE_ENV = "development";
        const { code } = babel.transform(inputCode, { plugins: [plugin] });
        expect(code).toMatchSnapshot();
    });

    it("prod", () => {
        // input
        process.env.NODE_ENV = "production";
        const { code } = babel.transform(inputCode, { plugins: [plugin] });
        expect(code).toMatchSnapshot();
    });
});