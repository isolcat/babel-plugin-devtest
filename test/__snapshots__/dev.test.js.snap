// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-dev-debug dev 1`] = `
"const a = 10;
const b = 20;
console.log(a + b);
if (DEBUG) {
  console.log("Hello World");
}"
`;

exports[`babel-plugin-dev-debug prod 1`] = `
"const a = 10;
const b = 20;
console.log(a + b);
if (DEBUG) {
  console.log("Hello World");
}"
`;
